client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -56,38 +56,43 @@ const HostingFeatures = () => {
 			? `/hosting-config/${ siteId }`
 			: `/overview/${ siteId }`
 	);
+	const hasRedirectQueryParam = searchParams.has( 'redirect_to' );
```

Comment:
Because the redirect behavior here is so nuanced, and break a site if the button is clicked mid-transfer (as it can does today), it might be good to add a unit test to confirm that `redirect_to` params are triggering the desired redirect.

Reply:
> break a site if the button is clicked mid-transfer

Do you mean the `Activate now` button? But that button won't render while the transfer is in progress, right, @nightnei?

Testing is good, but I don't think it's necessarily a blocker for this PR (I would be more adamant on this point if we knew what caused the underlying issue that makes it, so we need the delay for the regular redirect).

Reply:
> Do you mean the Activate now button? But that button won't render while the transfer is in progress...

Correct. I was thinking of what would happen if a regression ever led to that button rendering again, as it did before @nightnei made the changes to hide it during a transfer.